src/mmrelay/plugin_loader.py

```
@@ -27,6 +170,124 @@ def _reset_caches_for_tests():
+
+    try:
+        in_pipx = any(
+            key in os.environ
+            for key in ("PIPX_HOME", "PIPX_LOCAL_VENVS", "PIPX_BIN_DIR")
+        )
+
+        if in_pipx:
+            logger.info("Installing requirements for plugin %s with pipx", repo_name)
+            pipx_path = shutil.which("pipx")
+            if not pipx_path:
+                raise FileNotFoundError("pipx executable not found on PATH")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)